File: LeetCode/Python/#551 Student Attendance Record I.py
class Solution(object):
    def checkRecord(self, s):
        """
        :type s: str
        :rtype: bool
        """
        res = {"L":0}
        s = s.upper()
        if s.count("A")>=2:
            return False
        for i in s:
            if i=="L":
                res["L"] = res.get("L",1)+1
                if res["L"]>=3:
                    return False
            else:
                res["L"] = 0
        return True


class Solution(object):
    def checkRecord(self, s):
        """
        :type s: str
        :rtype: bool
        """
        return not re.search('A.*A|LLL',s)

File: LeetCode/JavaScript/#551 Student Attendance Record I.js
/**
 * @param {string} s
 * @return {boolean}
 */
var checkRecord = function(s) {
    return !(/A.*A|LLL/.test(s));
};



var checkRecord = function(s) {
    var res = {"L":0,"A":0};
    var len = s.length;
    for(let i in s){
      if(s[i]==="L"){
        res["L"] +=1;
        if(res["L"]>=3){
          return false;
        }
      }else if(s[i]==="A"){
        res["L"] = 0;
        res["A"] +=1;
        if(res["A"]>=2){
          return false;
        }
      }else{
        res["L"] = 0;
      }
    }
    return true; 
};